evap/staff/tools.py

```diff
@@ -68,31 +69,95 @@ def delete_navbar_cache_for_users(users):
         cache.delete(key)
 
 
-def bulk_delete_users(request, username_file, test_run):
-    usernames = [u.decode().strip() for u in username_file.readlines()]
-    users = UserProfile.objects.exclude(username__in=usernames)
-    deletable_users = [u for u in users if u.can_be_deleted_by_manager]
-    users_to_mark_inactive = [u for u in users if u.is_active and not u.can_be_deleted_by_manager and u.can_be_marked_inactive_by_manager]
-
-    messages.info(request, _('The uploaded text file contains {} usernames. {} other users have been found in the database. '
-                           'Of those, {} will be deleted and {} will be marked inactive.')
-                  .format(len(usernames), len(users), len(deletable_users), len(users_to_mark_inactive)))
-    messages.info(request, mark_safe(_('Users to be deleted are:<br />{}')
-                  .format('<br />'.join([u.username for u in deletable_users]))))
-    messages.info(request, mark_safe(_('Users to be marked inactive are:<br />{}')
-                  .format('<br />'.join([u.username for u in users_to_mark_inactive]))))
+def find_matching_internal_user_for_email(request, email):
+    # for internal users only the part before the @ must be the same to match a user to an email
```

Comment:
is that "fuzzy match" something the importers should also do?

Reply:
I'd put it in the importers too.

Reply:
i find it actually strange to do the fuzzy match at all and would like to keep it only where absolutely needed. in the long run i hope we can completely remove it from the code when the users in our productive environment will get unified email addresses (which hopefully happens soon).

Reply:
Fair enough, I think that is reasonable. (Email-issue: #1374)